packages/cadl-ranch-specs/http/server/version/union/main.tsp

```diff
@@ -0,0 +1,66 @@
+import "@typespec/http";
+import "@typespec/versioning";
+import "@azure-tools/cadl-ranch-expect";
+import "@azure-tools/typespec-azure-core";
+
+using Azure.Core;
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.Versioning;
+
+@doc("Illustrates server with customer-defined and template-defined api versions.")
+@scenarioService("/server/version/union")
+@useDependency(Azure.Core.Versions.v1_0_Preview_1)
+namespace Server.Version.Union;
+
+@doc("Pet")
+@resource("Pet")
+model Pet {
+  @key
+  @visibility("read")
+  @doc("pet name")
+  name: string;
+}
+
+#suppress "@azure-tools/typespec-azure-core/use-standard-operations" "For testing purpose"
+@doc("Server with multiple api versions and operation with customer-defined api version.")
+@scenario
+@scenarioDoc("""
+Server with multiple api versions and operation with customer-defined api version.
+
+Expected path with api version parameter: /server/version/union/customerDefined?api-version=1.0.0
+""")
+@route("/customerDefined")
+@head
+op customerDefined(
+  @doc("Api version.")
+  @query
+  apiVersion: string
+): NoContentResponse;
+
+#suppress "@azure-tools/typespec-azure-core/use-standard-operations" "For testing purpose"
+@doc("Server with multiple api versions and operation with customer-defined api version.")
+@scenario
+@scenarioDoc("""
+Server with multiple api versions and operation with customer-defined api version.
+
+Expected path with api version parameter: /server/version/union/customerDefined?api-version=2.0.0
+""")
+@route("/customerDefinedWithDefault")
+@head
+op customerDefinedWithDefault(
+  @doc("Api version.")
+  @query
+  apiVersion: "2.0.0"
+): NoContentResponse;
+
+#suppress "@azure-tools/typespec-azure-core/use-standard-operations" "For testing purpose"
+@doc("Server with multiple api versions and operation with template-defined api version.")
+@scenario
+@scenarioDoc("""
+Server with multiple api versions and operation with template-defined api version.
+
+Expected path with api version parameter: /server/version/union/Pet/dog?api-version=1.0.0
+""")
+@delete
+op templateDefined is ResourceDelete<Pet>;
```

Comment:
what is this operation, does using azure.core here bring any value?

We should try to keep all azure things into the Azure folder.

Reply:
This case is for: when customer defined api-version and templated defined api-version are put together, there should be no issues.

Reply:
What is the relationship between azure core template api version and service api version? I wonder if service should define its own api version instead of reuse the template api version?

Reply:
Azure core template defines api version in its implementation. Then if user doesn't use template, he might define his own api version. Put it in another way, user just spread everything in the template, then there would be an apiversion paramter, right?

Reply:
>  Which from what I understand is you have 2 different operations with different versioning scheme.

I don't think we support 2 different versioning values in the same client. For the client generated from this spec, its `api-version` should be either `v1.1` or `v1.2`, and it applies to all the operations.

Because api-version is a special parameter, what I want to test is it is handled correctly no matter it is defined in azure.core, or in raw operation, or in both of them in the same spec. I don't care if its value is different or same (though it cannot be different).

Reply:
Though Azure.Core templates just render down to a simple operation, api-version is a such special parameter that we once not support this spec. Therefore it is valuable I think.

Reply:
so what you are trying to test here is the `Azure.Core.Foundations.ApiVersionParameter`, I don't think doing a mix bring much value, not really sure we are going to see that ever.

So few items:
1. Move the those test under Azure
2. The path one is good
3. Have a AzureCoreApiVersionParameter spec that use that reference with a custom operation. Or use `Azure.Core.Foundation.Operation` with empty objects. Don't include anything extra(no Pet)
4. The query one should be renamed to something like NonStandardQuery.